src/main/java/com/openlattice/shuttle/dates/JavaDateTimeHelper.java

```diff
@@ -1,81 +1,118 @@
 
-    public JavaDateTimeHelper(TimeZone tz, String... datePatterns) {
+    public JavaDateTimeHelper( Optional<TimeZone> tz, String... datePatterns ) {
         this.tz = tz;
         this.datePatterns = datePatterns;
     }
 
-    private boolean shouldIgnoreValue(String date) {
-        return date == null || StringUtils.isBlank(date) || date.equals("NULL");
+    private boolean shouldIgnoreValue( String date ) {
+        return StringUtils.isBlank( date ) || date.equals( "NULL" );
     }
 
-    public <R> R parseWithTwoDigitYearHandling(String date, BiFunction<String, DateTimeFormatter, R> parseFunction,
+    public <R> R parseWithTwoDigitYearHandling(
+            String date, BiFunction<String, DateTimeFormatter, R> parseFunction,
             BiFunction<R, String, R> postParseFunction ) {
-        if (shouldIgnoreValue(date)) {
+        if ( shouldIgnoreValue( date ) ) {
             return null;
         }
-        for ( String datePattern: datePatterns ){
+        for ( int i = 0; i < datePatterns.length; i++ ) {
+
             try {
-                DateTimeFormatter formatter = Cached.getDateFormatForString( datePattern );
+                DateTimeFormatter formatter = Cached.getDateFormatForString( datePatterns[ i ] );
                 R result = parseFunction.apply( date, formatter );
-                return postParseFunction.apply( result, datePattern );
-            } catch ( DateTimeParseException e) {
-                if ( datePattern.equals(datePatterns[datePatterns.length - 1]) ) {
-                    logger.error("Unable to parse date {}, please see debug log for additional information: {}.", date,e);
+                return postParseFunction.apply( result, datePatterns[ i ] );
+            } catch ( DateTimeParseException e ) {
+                if (i == datePatterns.length - 1) {
+                    logger.error( "Unable to parse date {}, please see debug log for additional information: {}.",
+                            date,
+                            e );
                 }
             } catch ( ExecutionException ex ) {
-                logger.error("ExecutionException loading pattern from cache", ex);
+                logger.error( "ExecutionException loading pattern from cache", ex );
             }
+
         }
         return null;
     }
 
-    public LocalDate parseDate(String date) {
+    public LocalDate parseDate( String date ) {
         return parseWithTwoDigitYearHandling( date, LocalDate::parse, ( ld, datePattern ) -> {
-            if (  checkDatePatternIsTwoDigitYear( datePattern ) ) {
+            if ( checkDatePatternIsTwoDigitYear( datePattern ) ) {
                 // TODO: break this out into its own transform that specifies the date boundaries for two-digit years
-                if ((ld.getYear() - LocalDate.now().getYear()) > 20) {
-                    ld = ld.withYear(ld.getYear() - 100);
+                if ( ( ld.getYear() - LocalDate.now().getYear() ) > Constants.DECADE_CUTOFF ) {
```

Comment:
There's only minor simplification that can be done here since this logic needs to be applied on different types, but you could do something like
```
private boolean shouldAdjustTwoDigitYear( String datePattern, int year ) {
  return checkDatePatternIsTwoDigitYear( datePattern ) && year - LocalDate.now().getYear() > Constants.DECADE_CUTOFF;
}
```
and then here:
```
return parseWithTwoDigitYearHandling( date, LocalDate::parse, ( ld, datePattern ) -> {
  if ( shouldAdjustTwoDigitYear( datePattern, ld.getYear() ) {
    ld = ld.withYear( ld.getYear() - 100 );
  }

  return ld;
}


(... line 105)

( odt, datePattern ) -> {
  if ( shouldAdjustTwoDigitYear( datePattern, odt.getYear() ) ) {
    odt = odt.withYear( odt.getYear() - 100 );
  }

  return odt;
}